client/components/embed-container/index.jsx

```diff
@@ -192,6 +196,34 @@ function embedSlideshow( domNode ) {
 	}
 }
 
+function embedCarousel( domNode ) {
+	debug( 'processing carousel for ', domNode );
+
+	const carouselItemsWrapper = domNode.querySelector( '.swiper-wrapper' );
+
+	// Inject the DotPager component.
+	if ( carouselItemsWrapper ) {
+		const carouselItems = Array.from( carouselItemsWrapper?.children );
+
+		if ( carouselItems ) {
```

Comment:
Nit: should we also require `carouselItems` to have a length?
```suggestion
		if ( carouselItems && carouselItems.length ) {
```